src/main/scala/faces/apps/RandomFaces.scala

```diff
@@ -100,27 +100,33 @@ object RandomFaces extends App {
             centered.camera.principalPoint.y + (2.0 * yTranslationDistribution()) / imageHeight)
         ))
 
-        // render image with or without background
-        val img = if(bg) {
-          require(helpers.loadBgs.nonEmpty, "no Background files with type " + cfg.backgrounds.bgType + " found in " + cfg.backgrounds.bgPath)
-          val rndBG = helpers.loadBgs(rnd.scalaRandom.nextInt(helpers.loadBgs.length))
-          val rndBGimg = PixelImageIO.read[RGBA](rndBG).get.resample(imageWidth, imageHeight)
-          helpers.renderer.renderImage(rps).zip(rndBGimg).map(p => if (p._1.a < 0.5) p._2 else p._1 )
-        }
-        else{
-          helpers.renderer.renderImage(rps)
-        }
+        val imageData =
+          for ((postfix, currentRenderer) <- helpers.renderingMethods) yield {
+            if (bg && postfix == "") { // only allow different backgrounds for standard renderings
+              require(helpers.loadBgs.nonEmpty, "no Background files with type " + cfg.backgrounds.bgType + " found in " + cfg.backgrounds.bgPath)
+              val rndBG = helpers.loadBgs(rnd.scalaRandom.nextInt(helpers.loadBgs.length))
+              val rndBGimg = PixelImageIO.read[RGBA](rndBG).get.resample(imageWidth, imageHeight)
+              (currentRenderer.renderImage(centered).zip(rndBGimg).map(p => if (p._1.a < 0.5) p._2 else p._1), postfix)
```

Comment:
Why do we use here `centered`? On line 97 there is a variable `rps` calculated from `centered` but never used. Is this intended? Can this variable be removed?